### PR TITLE
provide support for the external-cloud-provider relation

### PR DESCRIPTION
**Pull request by addyess**
do not merge until https://github.com/charmed-kubernetes/charm-lib-kubernetes-snaps/pull/7 is merged in

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 aiohttp>=3.7.4,<3.8.0
 charm-lib-contextual-status @ git+https://github.com/charmed-kubernetes/charm-lib-contextual-status
 charm-lib-interface-container-runtime @ git+https://github.com/charmed-kubernetes/charm-lib-interface-container-runtime
+charm-lib-interface-external-cloud-provider @ git+https://github.com/charmed-kubernetes/charm-lib-interface-external-cloud-provider
```
Comment (addyess) [inline requirements.txt:4]: revert this once the dependent PR is merged

File: requirements.txt [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
 charm-lib-interface-kube-dns @ git+https://github.com/charmed-kubernetes/charm-lib-interface-kube-dns
 charm-lib-interface-kubernetes-cni @ git+https://github.com/charmed-kubernetes/charm-lib-interface-kubernetes-cni
-charm-lib-kubernetes-snaps @ git+https://github.com/charmed-kubernetes/charm-lib-kubernetes-snaps
+charm-lib-kubernetes-snaps @ git+https://github.com/charmed-kubernetes/charm-lib-kubernetes-snaps@feature/cloud-integration
```
Comment (addyess) [inline requirements.txt:7]: revert this once the pending branch is merged

**Review by mateoflorido — APPROVED** [review]
LGTM!

**Comment by Cynerva** [comment]
:+1: Nice, looks good

**Pull request merged**